src/flib/pio_nf.F90

```diff
@@ -1709,7 +1709,7 @@ integer function def_var_chunking(file, vardesc, storage, chunksizes) result(ier
     type (var_desc_t), intent(in) :: vardesc
     integer, intent(in) :: storage
     integer, intent(in) :: chunksizes(:)
-    integer(C_INT) :: cchunksizes(PIO_MAX_VAR_DIMS)
+    integer(C_SIZE_T) :: cchunksizes(PIO_MAX_VAR_DIMS)
```

Comment:
Nice catch! Thanks!

Reply:
Actually, before I merge this, can you tell me about it?

Also, is there a test where this change is demonstrated?

Reply:
The C function takes type PIO_Offset * for the chunksizes which elsewhere in the code corresponds to a C_SIZE_T array type in the Fortran interfaces. Which makes sense - chunksize are array sizes (hence size_t).  

My reading of it was that this code had been copied from elsewhere (e.g. PIO_def_var) where the array argument (e.g. dimids) holds indices rather than sizes. This also accounts for the change on line 1727, the -1 is not needed since we are dealing with sizes.

Reply:
OK, I added a test and applied the suggested changes, and had to make a few more changes as well. ;-)

It was all merged as part of #1738